src/nodedef.cpp

```diff
@@ -1683,8 +1698,14 @@ bool NodeDefManager::nodeboxConnects(MapNode from, MapNode to,
 				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 				0, 0, 0, 0, 0, 0, 32, 16, 8, 4 // 32 - left
 				};
-			return (f2.connect_sides
-				& rot[(connect_face * 4) + (to.param2 & 0x1F)]);
+			if (f2.param_type_2 == CPT2_FACEDIR ||
+					f2.param_type_2 == CPT2_COLORED_FACEDIR) {
+				return (f2.connect_sides
+					& rot[(connect_face * 4) + (to.param2 & 0x1F)]);
+			} else  {
```

Comment:
Consider a comment or redundant `else if` for 4dir, in case of future paramtype2s